### ISSUE TITLE
How to stop `SFTTrainer` from auto tokenizing my messages ?

### ISSUE THREAD

**Issue by MohamedAliRashad** [huggingface/trl#2642]
I want to tokenize my text in a custom way in a custom data collator but for some reason i don't know the data keeps being auto tokenized.
I passed `processing_class=None` to stop this but nothing changed, how can i stop the auto tokenization process ?

**Comment by qgallouedec** [comment]
Can your custom way of tokenizing be done before training? That's usually a good practice not to tokenize on the fly.

**Comment by MohamedAliRashad** [comment]
@qgallouedec All i need to provide is `input_ids`, `attention_mask` and `labels` and that's it, correct ?

**Comment by qgallouedec** [comment]
`input_ids` should be enough

**Comment by wasifmasood** [comment]
was this question already answered? I am also stuff with this problem.

**Comment by MohamedAliRashad** [comment]
Yes
